sidekick/tests/test_views.py

```diff
@@ -554,3 +558,79 @@ def test_redirect_url(self):
         self.assertRedirects(
             response, "http://example.org", fetch_redirect_response=False
         )
+
+
+class LabelTurnConversationViewTests(APITestCase):
+    def test_auth_required(self):
+        """
+        Authorization is required to access the endpoint
+        """
+        url = reverse("label-turn-conversation", args=[1])
+        response = self.client.post(url)
+        self.assertEqual(response.status_code, status.HTTP_401_UNAUTHORIZED)
+
+    def test_permission_required(self):
+        """
+        The user needs the appropriate permission to access the endpoint
+        """
+        url = reverse("label-turn-conversation", args=[1])
+        user = get_user_model().objects.create_user("test")
+        self.client.force_authenticate(user)
+        response = self.client.post(url)
+        self.assertEqual(response.status_code, status.HTTP_403_FORBIDDEN)
+
+    def login_user(self):
+        user = get_user_model().objects.create_user("test")
+        permission = Permission.objects.get(name="Can label a Turn Conversation")
+        user.user_permissions.add(permission)
+        user.save()
+        self.client.force_authenticate(user)
+
+    def test_invalid_id(self):
+        """
+        If an Org with the specified ID doesn't exist in the database, we should return
+        a Not Found
+        """
+        url = reverse("label-turn-conversation", args=[1])
+        self.login_user()
+
+        response = self.client.post(
+            url,
+            {"contact": {"uuid": str(uuid4()), "urn": "whatsapp:27820001001"}},
+            format="json",
+        )
+        self.assertEqual(response.status_code, status.HTTP_404_NOT_FOUND)
+
+    def test_invalid_body(self):
+        """
+        If the body of the request is invalid, we should return a Bad Request error
+        """
+        org = Organization.objects.create()
```

Comment:
Shouldn't this throw an error? Most fields have `null=False, blank=False` set on the model. . .

Reply:
Apparently not?
```
>>> from sidekick.models import Organization
>>> org = Organization.objects.create()
>>> org.name
''
>>> org.url
''
>>> org.token
''
```

Reply:
I've switched to using the base class in these tests, and then using `self.org` as the organisation